### ISSUE TITLE
Let the spawn menu set a level tag 

### ISSUE THREAD

**Issue by DimasVoxel** [Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker#272]
### Is your feature request related to a problem (if not, explain why you think this feature should be added)? Please describe.

Currently when spawning objects using the spawn menu scripts and sounds get triggered even if the object isn't placed on its final position.  A way to detect the current state would be welcome 

### Describe the solution you'd like

A solution would be that a certain level tag will be set by the spawn menu script notifying the creation that it's currently in the process of being spawned. 

Necessary checks within the script can created to stop or play specific sounds during the spawning process. The T-Rex mod for example could have checks not to rawr while being spawned.

In addition to that some mods require specific collision filters to function properly. But since the spawn menu sets its own filters creations can brake sometimes. Having the possibility to detect that the creation was spawned will further help to have bug free mods

### Describe alternatives you've considered

I tried to write my own checks but none were consistent and precise enough for every situation

**Comment by nooitaf** [comment]
A quick workaround for the moment might be checking for this before you go forward with the script.. (untested)
```lua
if GetString("game.player.tool") == "none" then
  -- still spawning
else
  -- spawned
end
```
... as while spawning the spawn menu sets the tool to `none`

**Comment by DimasVoxel** [comment]
Yeah I've been thinking about that too. But I am afraid that it can lead to issues on certain maps. While very unlikely I believe it can still happen. I would rather wait for something less jank and more official
